app/lib/search/mem_index.dart

```diff
@@ -49,36 +50,34 @@ class InMemoryPackageIndex {
     }
     _updateOverallScores();
     _lastUpdated = clock.now().toUtc();
-    _overallOrderedHits = _rankWithComparator(_compareOverall,
+    _overallOrderedHits = _rankWithComparator(_packages, _compareOverall,
```

Comment:
Not sure: is there any benefit of extracting these methods outside of the class? It is not that we will reuse them in other places.

Reply:
This was more of a transitionary change to a world where a lot of the mutable state doesn't have to escape construction. But that proved too large, so I thought this initial change would still be an improvement for now while getting a bit closer.

I do think making it `static` helps make it clear (and enforces that) it's not impacting any state of the class.